ci: E2E tests failing due to problem loading Twitter widget

The [last nightly CI tests](https://github.com/HEPData/hepdata/actions/runs/3223679185) failed even after being restarted four times.  The [error message](https://github.com/HEPData/hepdata/actions/runs/3223679185/jobs/5283048402#step:14:622) comes from loading the Twitter widget on the homepage:

```
E       AssertionError: Errors in browser log:
[622](https://github.com/HEPData/hepdata/actions/runs/3223679185/jobs/5283048402#step:14:623)
E         SEVERE: https://platform.twitter.com/js/horizon_timeline.a7991bb824d62c8d5038ddd875db8389.js - Failed to load resource: the server responded with a status of 502 (Bad Gateway)
[623](https://github.com/HEPData/hepdata/actions/runs/3223679185/jobs/5283048402#step:14:624)
E       assert 1 == 0
[624](https://github.com/HEPData/hepdata/actions/runs/3223679185/jobs/5283048402#step:14:625)
E        +  where 1 = len([{'level': 'SEVERE', 'message': 'https://platform.twitter.com/js/horizon_timeline.a7991bb824d62c8d5038ddd875db8389.js - Failed to load resource: the server responded with a status of 502 (Bad Gateway)', 'source': 'network', 'timestamp': 1665500364925}])
```

The E2E tests seem to pass fine when run locally.  There have been [recent changes](https://twittercommunity.com/t/embedded-timelines-update-parameters-support/177112) to the Twitter widget and the latest version seems to not be compatible with our E2E tests.  Maybe we need to wait for the Twitter widget to load on the homepage before proceeding.  I'm going to first try updating the HTML snippet for the Twitter widget.

--- a/hepdata/config.py
+++ b/hepdata/config.py
@@ -266,6 +266,7 @@ TWITTER_HANDLE_MAPPINGS = {
     "bicep2": "@BICEPTWO",
     "xenon": "@Xenon1T",
     "microboone": "@Microboone",
+    "belle-II": "@belle2collab",
 }
 
 INVALID_DOI_TEMPLATE = "hepdata_theme/invalid_doi.html"

--- a/hepdata/modules/permissions/api.py
+++ b/hepdata/modules/permissions/api.py
@@ -147,16 +147,16 @@ def write_submissions_to_files():
     """Writes some statistics on number of submissions per Coordinator to files."""
 
     import csv
-    from datetime import date
+    from datetime import datetime
 
     # Open a CSV file to write the number of unfinished and finished submissions for each Coordinator.
-    csvfile = open('submissions_per_coordinator_{}.csv'.format(date.today()), 'w')
+    csvfile = open('submissions_per_coordinator_{}.csv'.format(datetime.utcnow().date()), 'w')
     writer = csv.writer(csvfile)
     writer.writerow(['user_id', 'user_email', 'collaboration', 'version',
                  'number_todo', 'number_finished'])
 
     # Open another CSV file to write the collaboration and date of each finished version 1 submission.
-    csvfile1 = open('submissions_with_date_{}.csv'.format(date.today()), 'w')
+    csvfile1 = open('submissions_with_date_{}.csv'.format(datetime.utcnow().date()), 'w')
     writer1 = csv.writer(csvfile1)
     writer1.writerow(['collaboration', 'publication_recid', 'inspire_id',
                   'created', 'last_updated'])

--- a/hepdata/modules/records/utils/records_update_utils.py
+++ b/hepdata/modules/records/utils/records_update_utils.py
@@ -173,7 +173,7 @@ def _get_time(date):
     date_is_int = (type(date) is int or type(date) is str and date.isdigit())
 
     if date_is_int:
-        specified_time = datetime.datetime.today() + datetime.timedelta(days=-abs(int(date)))
+        specified_time = datetime.datetime.utcnow() + datetime.timedelta(days=-abs(int(date)))
     else:
         specified_time = datetime.datetime.strptime(date, "%Y-%m-%d")
 

--- a/hepdata/modules/theme/templates/hepdata_theme/home.html
+++ b/hepdata/modules/theme/templates/hepdata_theme/home.html
@@ -35,6 +35,28 @@
 </script>
 {%- endblock json_ld %}
 
+{%- block header %}
+  {# https://developer.twitter.com/en/docs/twitter-for-websites/javascript-api/guides/set-up-twitter-for-websites #}
+  <script>window.twttr = (function(d, s, id) {
+    var js, fjs = d.getElementsByTagName(s)[0],
+      t = window.twttr || {};
+    if (d.getElementById(id)) return t;
+    js = d.createElement(s);
+    js.id = id;
+    js.src = "https://platform.twitter.com/widgets.js";
+    fjs.parentNode.insertBefore(js, fjs);
+
+    t._e = [];
+    t.ready = function(f) {
+      t._e.push(f);
+    };
+
+    return t;
+  }(document, "script", "twitter-wjs"));</script>
+  <meta name="twitter:dnt" content="on">
+  <meta name="twitter:widgets:csp" content="on">
+{% endblock header %}
+
 {%- block header_bars %}
 {%- endblock header_bars %}
 
@@ -155,11 +177,8 @@
             </div>
 
             <div class="twitter-area">
-                <a class="twitter-timeline" data-width="800" data-height="200" data-dnt="true"
-                   href="https://twitter.com/HEPData?ref_src=twsrc%5Etfw">
-                    Tweets by HEPData
-                </a>
-                <script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+              <a class="twitter-timeline" data-height="200"
+                 href="https://twitter.com/HEPData?ref_src=twsrc%5Etfw">Tweets by HEPData</a>
             </div>
 
             {% include "hepdata_search/modals/search_help.html" %}

--- a/hepdata/modules/theme/templates/hepdata_theme/home.html
+++ b/hepdata/modules/theme/templates/hepdata_theme/home.html
@@ -55,6 +55,7 @@
   }(document, "script", "twitter-wjs"));</script>
   <meta name="twitter:dnt" content="on">
   <meta name="twitter:widgets:csp" content="on">
+  <meta name="twitter:widgets:autoload" content="off">
 {% endblock header %}
 
 {%- block header_bars %}
@@ -176,7 +177,7 @@
 
             </div>
 
-            <div class="twitter-area">
+            <div class="twitter-area" id="twitter-feed">
               <a class="twitter-timeline" data-height="200"
                  href="https://twitter.com/HEPData?ref_src=twsrc%5Etfw">Tweets by HEPData</a>
             </div>
@@ -188,6 +189,9 @@
 
 {%- block javascript %}
   {{ webpack['hepdata-home-js.js'] }}
+  {% if not config['TESTING'] %}
+    <script>twttr.widgets.load(document.getElementById("twitter-feed"));</script>
+  {%  endif %}
 {% endblock %}
 
 {%- block trackingcode %}

--- a/hepdata/modules/theme/templates/hepdata_theme/pages/about.html
+++ b/hepdata/modules/theme/templates/hepdata_theme/pages/about.html
@@ -242,6 +242,11 @@
 
             <ul class="talks">
                 {% with talks = [
+                {'meeting': 'Belle II Data Preservation Workshop',
+                'location': 'Roma Tre University, Rome',
+                'speaker': 'Graeme Watt',
+                'date': 'Oct 2022',
+                'link': 'https://indico.belle2.org/event/7653/contributions/44066/attachments/18872/28052/watt_hepdata_oct2022.pdf'},
                 {'meeting': 'STRONG-2020 meeting on database for hadronic cross sections',
                  'location': 'Virtual Zoom',
                  'speaker': 'Graeme Watt',

--- a/hepdata/version.py
+++ b/hepdata/version.py
@@ -28,4 +28,4 @@ This file is imported by ``HEPData.__init__``,
 and parsed by ``setup.py``.
 """
 
-__version__ = "0.9.4dev20221011"
+__version__ = "0.9.4dev20221012"

--- a/hepdata/version.py
+++ b/hepdata/version.py
@@ -28,4 +28,4 @@ This file is imported by ``HEPData.__init__``,
 and parsed by ``setup.py``.
 """
 
-__version__ = "0.9.4dev20220926"
+__version__ = "0.9.4dev20221011"

--- a/setup.py
+++ b/setup.py
@@ -35,7 +35,7 @@ history = open('CHANGES.rst').read()
 
 tests_require = [
     'pytest>=6.0.2',
-    'pytest-cov>=2.9.0',
+    'pytest-cov>=2.9.0,<4.0.0',
     'pytest-flask>=1.0.0',
     'pytest-mock>=3.1.0',
     'pytest-timeout>=1.4.2',

--- a/tests/dashboard_test.py
+++ b/tests/dashboard_test.py
@@ -476,7 +476,7 @@ def test_submissions_csv(app, admin_idx, load_default_data, identifiers):
         csv_lines = csv_data.splitlines()
         assert len(csv_lines) == 3
         assert csv_lines[0] == 'hepdata_id,version,url,inspire_id,arxiv_id,title,collaboration,creation_date,last_updated,status,uploaders,reviewers'
-        today = datetime.date.today().isoformat()
+        today = datetime.datetime.utcnow().date().isoformat()
         assert csv_lines[1] == f'16,1,http://localhost/record/16,1245023,arXiv:1307.7457,High-statistics study of $K^0_S$ pair production in two-photon collisions,Belle,{today},2013-12-17,finished,,'
         assert csv_lines[2] == f'1,1,http://localhost/record/1,1283842,arXiv:1403.1294,Measurement of the forward-backward asymmetry in the distribution of leptons in $t\\bar{{t}}$ events in the lepton$+$jets channel,D0,{today},2014-08-11,finished,,'
 

